### PR TITLE
feat!: Support slo-generator config v2 format (core changes)

### PR DESCRIPTION
**Pull request by ocervell**
This PR adds support for the new slo-generator configuration v2 format, which basically:
* Upgrades the SLO config to a k8s-compatible one, and field names that make more sense
* Adds a new slo-generator config, that is a super-set of the existing error budget policy config (which is now obsolete)

### CODE REVIEW

File: slo_generator/report.py [hunk 1]
```diff
@@ -158,41 +174,52 @@ def build(self, step, data):
...
-        cls = cfg.get('class')
-        method = cfg.get('method')
-        excluded_keys = ['class', 'method', 'measurement']
-        backend_cfg = {k: v for k, v in cfg.items() if k not in excluded_keys}
-        instance = utils.get_backend_cls(cls)(client=client, **backend_cfg)
+        cls_name = backend.get('class')
+        method = config['spec']['method']
+        excluded_keys = ['class', 'service_level_indicator', 'name']
+        backend_cfg = {
+            k: v for k, v in backend.items() if k not in excluded_keys
```
Comment (lvaylet) [inline slo_generator/report.py:196]: Cosmetics. Split over 3 separate lines to be consistent with the rest of the codebase?

**Review by lvaylet — COMMENTED** [review]
One typo. Three minor cosmetic improvements.

File: slo_generator/backends/dynatrace.py [hunk 2]
```diff
@@ -83,17 +83,15 @@ def threshold(self, timestamp, window, slo_config):
...
         start = (timestamp - window) * 1000
         end = timestamp * 1000
         query_valid = measurement['query_valid']
         threshold = measurement['threshold']
         good_below_threshold = measurement.get('good_below_threshold', True)
         response = self.query(start=start, end=end, **query_valid)
         LOGGER.debug(f"Result valid: {pprint.pformat(response)}")
-        return DynatraceBackend.count_threshold(response,
-                                                threshold,
+        return DynatraceBackend.count_threshold(response, threshold,
```
[the remark this answers is not in the document]
Comment (ocervell) [reply slo_generator/backends/dynatrace.py:94]: This is automatically formatted by `yapf` using `--style {based_on_style: google, column_limit: 80, indent_width: 4}`

File: slo_generator/exporters/base.py [hunk 3]
```diff
@@ -87,32 +94,33 @@ class `export_metric` method.
                 }
             if metric_cfg['name'] == 'error_budget_burn_rate':
                 metric_cfg = MetricsExporter.use_deprecated_fields(
-                    config=config,
-                    metric=metric_cfg)
+                    config=config, metric=metric_cfg)
```
[the remark this answers is not in the document]
Comment (ocervell) [reply slo_generator/exporters/base.py:97]: ditto

File: slo_generator/report.py (hunk 1 above)
Comment (ocervell) [reply slo_generator/report.py:196]: ditto

**Pull request merged**